internal/internal_workflow_client_test.go

```diff
@@ -1063,15 +1064,19 @@ func (s *workflowClientTestSuite) TestSignalWithStartWorkflow_Error() {
 	signalInput := []byte("my signal input")
 	options := StartWorkflowOptions{}
 
-	resp, err := s.client.SignalWithStartWorkflow(context.Background(), workflowID, signalName, signalInput,
-		options, workflowType)
+	// Pass a context with a deadline so error retry doesn't take forever
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	resp, err := s.client.SignalWithStartWorkflow(ctx, workflowID, signalName, signalInput, options, workflowType)
```

Comment:
> because it retries up to the timeout, and it's not test-injectable.
yeah unfortunately. 
I was about to make the retry policy overridable but went this ctx deadline route. 
Overall test duration reduced as well because it was spending 1m retrying.